### PR TITLE
refactor / move base class to core

### PR DESCRIPTION
**Pull request by yzhang1994**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

- Related Clubhouse Story: https://app.clubhouse.io/coinalpha/story/3883/repo-organization

**A description of the changes proposed in the pull request**:
- Move TimeIterator, Pubsub, Clock, etc to hummingbot/core

### CODE REVIEW

**Comment by martinkou** [comment]
Also, you'll need to create a corresponding PR in hummingsim.

**Comment by yzhang1994** [comment]
I've included the cpp files. Will submit a PR to hummingsim soon

File: wings/tracker/ddex_active_order_tracker.pyx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # distutils: language=c++
-# distutils: sources=wings/cpp/OrderBookEntry.cpp
+# distutils: sources=hummingbot/core/cpp/orderBookEntry.cpp
```
Comment (martinkou) [inline wings/tracker/ddex_active_order_tracker.pyx]: Typo here, which killed the compilation.

Comment (yzhang1994) [reply wings/tracker/ddex_active_order_tracker.pyx]: fixed

**Comment by yzhang1994** [comment]
@martinkou I created PR in to hummingsim as well

**Pull request merged**